src/devices/Nrf24l01/Nrf24l01.cs

```diff
@@ -0,0 +1,773 @@
+        public DataRate DataRate { get => ReadDataRate(); set => SetDataRate(value); }
+
+        /// <summary>
+        /// nRF24L01 Power Mode
+        /// </summary>
+        public PowerMode PowerMode { get => ReadPowerMode(); set => SetPowerMode(value); }
+
+        /// <summary>
+        /// nRF24L01 Working Mode
+        /// </summary>
+        public WorkingMode WorkingMode { get => ReadWorkwingMode(); set => SetWorkingMode(value); }
+
+        /// <summary>
+        /// nRF24L01 Pipe 0
+        /// </summary>
+        public Nrf24l01Pipe Pipe0 { get; private set; }
+
+        /// <summary>
+        /// nRF24L01 Pipe 1
+        /// </summary>
+        public Nrf24l01Pipe Pipe1 { get; private set; }
+
+        /// <summary>
+        /// nRF24L01 Pipe 2
+        /// </summary>
+        public Nrf24l01Pipe Pipe2 { get; private set; }
+
+        /// <summary>
+        /// nRF24L01 Pipe 3
+        /// </summary>
+        public Nrf24l01Pipe Pipe3 { get; private set; }
+
+        /// <summary>
+        /// nRF24L01 Pipe 4
+        /// </summary>
+        public Nrf24l01Pipe Pipe4 { get; private set; }
+
+        /// <summary>
+        /// nRF24L01 Pipe 5
+        /// </summary>
+        public Nrf24l01Pipe Pipe5 { get; private set; }
+        #endregion
+
+        #region SpiSettings
+        /// <summary>
+        /// NRF24L01 SPI Clock Frequency
+        /// </summary>
+        public const int SpiClockFrequency = 2000000;
+
+        /// <summary>
+        /// NRF24L01 SPI Mode
+        /// </summary>
+        public const SpiMode SpiMode = System.Device.Spi.SpiMode.Mode0;
+        #endregion
+
+        /// <summary>
+        /// Creates a new instance of the nRF24L01.
+        /// </summary>
+        /// <param name="sensor">The communications channel to a device on a SPI bus</param>
+        /// <param name="ce">CE Pin</param>
+        /// <param name="irq">IRQ Pin</param>
+        /// <param name="packetSize">Receive Packet Size</param>
+        /// <param name="channel">Working Channel</param>
+        /// <param name="outputPower">Output Power</param>
+        /// <param name="dataRate">Send Data Rate</param>
+        /// <param name="pinNumberingScheme">Pin Numbering Scheme</param>
+        public Nrf24l01(SpiDevice sensor, int ce, int irq, byte packetSize, byte channel = 2,
+            OutputPower outputPower = OutputPower.N00dBm, DataRate dataRate = DataRate.Rate2Mbps, PinNumberingScheme pinNumberingScheme = PinNumberingScheme.Logical)
+        {
+            _sensor = sensor;
+            _ce = ce;
+            _irq = irq;
+            PacketSize = packetSize;
+
+            Initialize(pinNumberingScheme, outputPower, dataRate, channel);
+            InitializePipe();
+        }
+
+        /// <summary>
+        /// Send
+        /// </summary>
+        /// <param name="data">Send Data</param>
+        public void Send(byte[] data)
+        {
+            SetWorkingMode(WorkingMode.TX);
+            Thread.Sleep(4);
```

Comment:
how did you pick the values 4, 1, 10, etc?

Reply:
Note that sleeping for this amount of time isn't really possible, particularly on Windows (12ms+ is likely what you'll get). I'll add some helpers to \Common that will allow you to delay more appropriately as we're hitting this a ton.  I'll try and get these up today.

Reply:
@ZhangGaoxing good questions here - delay helpers also merged: https://github.com/dotnet/iot/blob/master/src/devices/Common/System/Device/DelayHelper.cs

Reply:
Actually, it doesn't make any special sense. I just don't want the program to go too fast. Now I change these values to 1.  I don't think DelayHelper is needed because these are not exact values.